lib/string_decoder.js

```diff
@@ -2,224 +2,203 @@
 
 const Buffer = require('buffer').Buffer;
 
-function assertEncoding(encoding) {
-  // Do not cache `Buffer.isEncoding`, some modules monkey-patch it to support
-  // additional encodings
-  if (encoding && !Buffer.isEncoding(encoding)) {
-    throw new Error('Unknown encoding: ' + encoding);
+// Do not cache `Buffer.isEncoding` when checking encoding names as some
+// modules monkey-patch it to support additional encodings
+function normalizeEncoding(enc) {
```

Comment:
Perhaps. I was working on some Buffer changes that might end up doing something similar.